cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -369,11 +369,11 @@ func StartControllers(s *options.CMServer, kubeconfig *restclient.Config, rootCl
 	// Find the list of namespaced resources via discovery that the namespace controller must manage
 	namespaceKubeClient := client("namespace-controller")
 	namespaceClientPool := dynamic.NewClientPool(restclient.AddUserAgent(kubeconfig, "namespace-controller"), restMapper, dynamic.LegacyAPIPathResolverFunc)
-	groupVersionResources, err := namespaceKubeClient.Discovery().ServerPreferredNamespacedResources()
+	deletableGroupVersionResources, err := namespaceKubeClient.Discovery().ServerPreferredNamespacedResourcesWithVerbs("delete")
```

Comment:
Got protobuf serializer working. Next: ugorji json decoding only guarantees len(in) = len(out) for slices, in other words: does not preserve the `[]`-`nil` distinction.